app/models/valid_datacite.rb

```diff
@@ -0,0 +1,141 @@
+      # @param dates [Array<Date>] different dates relevant to the work.
+      # @param language [String, nil] Primary language of the resource: an IETF BCP 47, ISO 639-1 language code.
+      # @param resource_type [ResourceType, nil] the type of the resource
+      # @param alternate_identifiers [Array<AlternateIdentifier>] an identifier or identifiers other than the primary {Identifier} applied to the resource being registered.
+      # @param related_identifiers [Array<RelatedIdentifier>] identifiers of related resources.
+      # @param sizes [Array<String>] unstructured size information about the resource.
+      # @param formats [Array<String>] technical format of the resource, e.g. file extension or MIME type.
+      # @param version [String] version number of the resource.
+      # @param rights_list [Array<Rights>] rights information for this resource.
+      # @param descriptions [Array<Description>] all additional information that does not fit in any of the other categories.
+      # @param geo_locations [Array<GeoLocations>] spatial region or named place where the data was gathered or about which the data is focused.
+      @mapping = Datacite::Mapping::Resource.new(
+        identifier: datacite_identifier(@metadata_from_form["identifier"]),
+        creators: datacite_creators,
+        titles: datacite_titles,
+        publisher: datacite_publisher(@metadata_from_form["publisher"]),
+        publication_year: @metadata_from_form["publication_year"],
+        # subjects: [],
+        # contributors: [],
+        # dates: [],
+        # language: nil,
+        # funding_references: [],
+        resource_type: datacite_resource_type(@metadata_from_form["resource_type"])
+        # alternate_identifiers: [],
+        # related_identifiers: [],
+        # sizes: [],
+        # formats: [],
+        # version: nil,
+        # rights_list: [],
+        # descriptions: [],
+        # geo_locations: []
+      )
+      @mapping
+    end
+
+    def to_xml
+      datacite_mapping.write_xml
+    end
+
+    ##
+    # For each creator, make a Datacite::Mapping::Creator object
+    # TODO: This class has no field for author order. We need a place to record that.
+    def datacite_creators
+      creator_array = []
+      @metadata_from_form["creators"].each do |creator|
+        creator_array << Datacite::Mapping::Creator.new(
+          name: creator["value"],
+          given_name: creator["given_name"],
+          family_name: creator["family_name"],
+          affiliations: creator["affiliations"]
+        )
+      end
+      creator_array
+    end
+
+    ##
+    # For each title, make a Datacite::Mapping::Title object
+    # TODO: Add qualifiers for different kinds of titles
+    def datacite_titles
+      title_array = []
+      @metadata_from_form["titles"].each do |title|
```

Comment:
I would prefer map here to
```suggestion
      @metadata_from_form["titles"].map do |title|
```